parser.c

```diff
@@ -60,7 +63,7 @@ static int parse_rw_string(struct tinyiiod *iiod, char *str, bool write)
 	}
 
 	ptr = strchr(str, ' ');
-	if (!!ptr ^ write)
+	if ((!!ptr )^ write)
```

Comment:
this should also be it's own commit ; and have a detailed explanation about it;
the reasoning is that it touches on library parts of libtinyiiod, and these should be explained when changing;
this is to avoid bad practices of changing things without any reasoning [even when the reason could be obvious]